### DISCUSSION TITLE
No Input Features & Error

### DISCUSSION THREAD

**Discussion by SamWright822** [e3nn/e3nn#336] [Q&A] [answered]
Hi!

I am wanting to train a model to predict 3D coordinates of 12 particles from their starting coordinates. I do not have input features. I think then for irreps_in I can put None? I'm not sure about this.

Also, I am trying to call my model (I used the e3nn.nn.models.gate_points_2101 Network)

I wanted to do a test quickly with my the first frame of my data (shape (12,3)), but I am having issues. I currently have None in the irreps_in place, so that may be why. 

After initializing all of the model parameters, I call the following:

`model = Network(**model_kwargs)`
`yhat = model(data)`

and I'm getting this error: `Tensor.__contains__ only supports Tensor or scalar, but you passed in a <class 'str'>.`

even though my data is type: `<class 'torch.Tensor'>`

I appreciate any help :)
Thanks!

**Comment by mariogeiger** [comment, marked as answer]
Looks like there is somewhere in the code something like `x in y` with `x` of type str and `y` of type tensor. You assumed the problem comes from the type of `x` but I think it probably comes from the type of `y`

**Reply by SamWright822** [reply]
I think I see what you mean. Thank you.

**Reply by SamWright822** [reply]
To be clear, it is acceptable to put `None` for the input features? (irreps_in)

**Reply by mariogeiger** [reply]
Yes. What is the error message you are getting?

**Reply by mariogeiger** [reply]
Maybe if you can make a minimal code that produce the error you get I will be able to better help you

**Reply by SamWright822** [reply]
For sure, thank you. I have tried this several ways, but here is a code snippet. My input is a frame (xyz coordinates for 12 points), and my output is the same:

```
train_x = torch.from_numpy(training_set)
train_y = torch.from_numpy(training_labels)

from e3nn.nn.models.gate_points_2101 import Network
from e3nn import o3

model_kwargs = {
    "irreps_in": None, 
    "irreps_hidden": o3.Irreps("5x0e + 5x0o + 5x1e + 5x1o"),
    "irreps_out": "12x1o", #12 vectors out
    "irreps_node_attr": None, 
    "irreps_edge_attr": o3.Irreps.spherical_harmonics(3), 
    "layers": 3, 
    "max_radius": 3.5,
    "number_of_basis": 10,
    "radial_layers": 1,
    "radial_neurons": 128,
    "num_neighbors": 11, 
    "num_nodes": 12,
    "reduce_output": False, 
}

model = Network(**model_kwargs)
model.to('cuda')

```

Then, as a test, I try to run my model with one frame:

```

output = model(train_x[0][0])

```

I have also tried to turn my data into a pytorch geometric dataset, which I run on all of my data (2048 frames):

```
ys = torch.from_numpy(labels)
feat = torch.from_numpy(features)

traj_data = torch_geometric.data.Data(pos=feat.to(torch.float32), x=ys.to(torch.float32))

output = model(traj_data.pos)

```

Both approaches give me the following error: 

``
RuntimeError: Tensor.__contains__ only supports Tensor or scalar, but you passed in a <class 'str'>.
``

**Reply by mariogeiger** [reply]
This architecture is made for point cloud data.
When you call the model `model(...)` you need to pass it a dictionary.
Try with

```python
model({'pos': train_x[0][0]})
```

**Reply by mariogeiger** [reply]
The `forward` function (called when executing the model) is defined here:
https://github.com/e3nn/e3nn/blob/main/e3nn/nn/models/gate_points_2101.py#L284-L295
